packages/app/src/locale/use-lokalize-text.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  createFlattenTexts,
+  parseLocaleTextDocument,
+} from '@corona-dashboard/common';
+import '@reach/combobox/styles.css';
+import { MutationEvent } from '@sanity/client';
+import css from '@styled-system/css';
+import { flatten, unflatten } from 'flat';
+import { debounce } from 'lodash';
+import set from 'lodash/set';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import { Subscription } from 'rxjs';
+import styled from 'styled-components';
+import { isDefined } from 'ts-is-present';
+import DatabaseIcon from '~/assets/database.svg';
+import { getClient } from '~/lib/sanity';
+import { LanguageKey, languages } from '~/locale';
+import { LokalizeText } from '~/types/cms';
+import { useHotkey } from '~/utils/hotkey/use-hotkey';
+import { useIsMountedRef } from '~/utils/use-is-mounted-ref';
+
+/**
+ * This hook will return an object which contains all lokalize translations.
+ *
+ * Additionally, with `enableHotReload: true` it can also connect with Sanity
+ * to receive real-time (hot) updates of draft translations.
+ *
+ * If enabled a tiny button will be rendered at the bottom-right which is only
+ * visible on hover.
+ *
+ * This button toggles though three states:
+ *
+ * - undefined (gray): app is rendered with translations from filesystem
+ * - live (green): app is listening to real-time sanity updates
+ * - paths (blue): app is displaying the pathnames of every key instead of the value
+ *
+ * The live/paths state can also be toggled with shortkey shift+t.
+ */
+export function useLokalizeText(
+  locale: LanguageKey,
+  { enableHotReload }: { enableHotReload?: boolean }
+) {
+  const isMountedRef = useIsMountedRef();
+  const [displayMode, setDisplayMode] = useState<'path' | 'live'>();
+
+  useHotkey(
+    'shift+t',
+    () => setDisplayMode((x) => (x === 'path' ? 'live' : 'path')),
+    {
+      isDisabled: !(enableHotReload && isDefined(displayMode)),
+      disableTextInputs: true,
+    }
+  );
+
+  const [text, setText] = useState(languages[locale]);
+  const textRef = useRef(text);
+  useEffect(() => {
+    textRef.current = text;
+  }, [text]);
+
+  useEffect(() => {
+    if (!enableHotReload || !isDefined(displayMode)) {
+      setText(languages[locale]);
+      return;
+    }
+
+    const setTextDebounced = debounce(setText, 500);
+    const query = `*[_type == 'lokalizeText']`;
+
+    let subscription: Subscription | undefined;
+
+    getClient().then(async (client) => {
+      await client
+        .fetch(query)
+        .then((documents: LokalizeText[]) => {
+          if (!isMountedRef.current) return;
+          const flattenTexts = createFlattenTexts(documents)[locale];
+          setText(() => unflatten(flattenTexts, { object: true }));
+        })
+        .catch((err) => console.error(err));
```

Comment:
Not sure why you are mixing async/await with promises here. I think it is cleaner to stick with async/await where possible.

```suggestion
      const documents: LokalizeText[] = await client.fetch(query)
      
      if (!isMountedRef.current) return;
          
      const flattenTexts = createFlattenTexts(documents)[locale];
      setText(() => unflatten(flattenTexts, { object: true }));
```

I think the catch can be moved to the getClient promise chain, because there it seems to be missing, and it would still print any error that gets thrown here.

Reply:
ah of course.. leftover from some refactoring